swap/protocol.go

```diff
@@ -73,25 +73,38 @@ func (s *Swap) Stop() error {
 
 func (s *Swap) verifyHandshake(msg interface{}) error {
 	handshake, ok := msg.(*SwapHandshakeMsg)
-	var empty common.Address
-	if !ok || handshake.Beneficiary == empty {
+	if !ok || (handshake.ContractAddress == common.Address{}) {
 		return ErrEmptyAddressInSignature
 	}
+
+	err := s.verifyContract(context.TODO(), handshake.ContractAddress)
```

Comment:
just write `return sp.Send(ctx, &ConfirmMsg{})` much cleaner, the `protocols` package will wrap this anyway in a failed handshake error